staging/src/k8s.io/apimachinery/pkg/util/intstr/intstr.go

```diff
@@ -151,6 +151,25 @@ func ValueOrDefault(intOrPercent *IntOrString, defaultValue IntOrString) *IntOrS
 	return intOrPercent
 }
 
+func GetIntFromIntOrPercent(intOrPercent *IntOrString, total int, roundUp bool) (int, error) {
+	if intOrPercent == nil {
+		return 0, errors.New("nil value for IntOrString")
+	}
+	value, isPercent, err := getIntOrPercentValueFromIntOrString(intOrPercent)
+	if err != nil {
+		return 0, fmt.Errorf("invalid value for IntOrString: %v", err)
+	}
+	if isPercent {
+		if roundUp {
+			value = int(math.Ceil(float64(value) * (float64(total)) / 100))
+		} else {
+			value = int(math.Floor(float64(value) * (float64(total)) / 100))
+		}
+	}
+	return value, nil
+}
+
+//Deprecated: use GetIntFromIntOrPercent
 func GetValueFromIntOrPercent(intOrPercent *IntOrString, total int, roundUp bool) (int, error) {
```

Comment:
we want to limit the bad code to this public method... I would recommend the following:

1. inline the old `getIntOrPercentValue` definition into `GetValueFromIntOrPercent` (so we contain the bad behavior to a single method)
2. fix `getIntOrPercentValue` to have the behavior we want
3. add a new public method which calls to `getIntOrPercentValue` (a clear name will be tricky to choose), with godocs clearly explaining the behavior for int values, numeric string values, percent string values, and malformed string values
4. mark `GetValueFromIntOrPercent` as deprecated (as you've done) and make sure there are no in-tree calls to it
5. add godocs to `GetValueFromIntOrPercent` explaining in detail the problematic behavior and the reason to switch to the new public method

Reply:
Is GetIntFromIntOrPercent a good name for the new method?

Reply:
not sure... `GetScaledValueFromIntOrPercent` might be a little clearer?

Reply:
GetValueFromIntOrPercentEx
GetValueFromIntOrPercentSafe
GetValueFromIntOrPercentWithErrorChecking

Reply:
Went with GetScaledValueFromIntOrPercent as that's effectively what the function does but as @liggitt said naming the new version of a function is tricky.

Reply:
"Scaled" is a very confusing name for this function. It isn't scaling anything.